// taobao/index.js
function Guo() {
    this.content = {};
}

Guo.prototype.append = function(Ingredients) {
    console.log(`锅中加入了${Ingredients?.name}`);
    this.content[Ingredients.name] = Ingredients;
}

Guo.prototype.fire = function() {
    console.log('锅中开始加热');
    if (this.content['水']) {
        this.content['水'].temperature = 100;
    }
}

Guo.prototype.bigFire = function() {
    console.log('大火收汁');
}

Guo.prototype.clear = function() {
    console.log('锅中内容被清空');
    this.content = {};
}


function ShaoshuiHu() {
    this.content = {};
}

ShaoshuiHu.prototype.fire = function() {
    console.log('烧水壶开始加热');
    if (this.content['水']) {
        this.content['水'].temperature = 100;
    }
}

ShaoshuiHu.prototype.append = function(Ingredients) {
    console.log(`烧水壶中加入了${Ingredients?.name}`);
    this.content[Ingredients.name] = Ingredients;
}

function Dao() {

}

Dao.prototype.cut = function(Ingredients) {
    console.log(`用刀切了${Ingredients?.name}`);
}

function Chanzi() {

}

Chanzi.prototype.striFry = function(time) {
    console.log(`用铲子翻炒${time}分钟`);
}


module.exports = {
    Guo,
    ShaoshuiHu,
    Dao,
    Chanzi
}
